apps/examples/dnsclient_test/dnsclient_main.c

```diff
@@ -159,12 +159,12 @@ int dnsclient_main(int argc, FAR char *argv[])
 
 	memset(hostname, 0x00, CONFIG_NETDB_DNSCLIENT_NAMESIZE);
 
-	if (strlen(argv[1]) > CONFIG_NETDB_DNSCLIENT_NAMESIZE) {
-		printf("dnsclient : length of hostname has to lower than %d\n", CONFIG_NETDB_DNSCLIENT_NAMESIZE);
+	if (strlen(argv[1]) > CONFIG_NETDB_DNSCLIENT_NAMESIZE - 1) {
+		printf("dnsclient : length of hostname has to lower than or equal to %d\n", CONFIG_NETDB_DNSCLIENT_NAMESIZE - 1);
 		return -1;
 	}
 
-	strncpy(hostname, argv[1], CONFIG_NETDB_DNSCLIENT_NAMESIZE);
+	strncpy(hostname, argv[1], CONFIG_NETDB_DNSCLIENT_NAMESIZE - 1);
```

Comment:
I think these statements should be
`if (strlen(argv[1]) > CONFIG_NETDB_DNSCLIENT_NAMESIZE - 1) {` and
`strncpy(hostname, argv[1], CONFIG_NETDB_DNSCLIENT_NAMESIZE);`
for NULL termination

Reply:
I think it doesn't matter.
However, for readability I use CONFIG_NETDB_DNSCLIENT_NAMESIZE - 1 instead of CONFIG_NETDB_DNSCLIENT_NAMESIZE.